evmos/ethermint
# release: v0.20-rc3 changelog

This PR aims to merge all features from `main` into the release branch history.

- deps(sdk): bump to v0.46.4 (#1423)
- imp: reduce integration test block time to 2s (#1428)
- build(deps): bump github.com/onsi/gomega from 1.23.0 to 1.24.0 (#1429)
- fix(rpc): different result from `eth_getProof` comparing with Ethereum (#1431)
- Refactor EIP-712 signature verification (#1397)
- fix: build test on mac by updating to python3.10 (#1437)
- build(deps): bump loader-utils from 1.4.0 to 1.4.1 in /tests/solidity (#1445)
- imp(evm): improve performance of EstimateGas (#1444)
- fix(rpc): decode `finalized` block number (#1442)
- build(deps): bump github.com/onsi/gomega from 1.24.0 to 1.24.1 (#1449)
- build(deps): bump github.com/spf13/viper from 1.13.0 to 1.14.0 (#1439)
- fix: unstable tx_priority test (#1440)
- build(deps): bump github.com/cosmos/ibc-go/v5 from 5.0.1 to 5.1.0 (#1450)
- build(deps): bump golangci/golangci-lint-action from 3.3.0 to 3.3.1 (#1454)
- refactor(all): refactor errors import to use cosmossdk.io (#1456)
- build(deps): bump alpine from 3.16.2 to 3.16.3 (#1453)
- Empty KV gas config (#1460)
- test: remove unused integration tests (#1462)
- fix: remove e2e github action (#1463)
- build(deps): bump loader-utils from 1.4.1 to 1.4.2 in /tests/solidity (#1464)
- chore (deps): Update geth version to v1.10.25 (#1413)
- Add EIP-712 encoding support type for any array (#1430)
- fix: extend geth config on integration tests (#1467)
- tests: Add unit tests for rpc client endpoints (#1409)
- Refactor to omit empty optionals from EIP-712 type generation (#1459)
- fix: protogen scripts were not correctly implemented (#1466)
- added gas consumption test (#1474)
- build(deps): bump github.com/cosmos/cosmos-proto (#1475)
- fix: chain-id in grpc query is not initialized without abci event (#1405)
- fix(evm): Simplify Gas Math (#1452)
- imp(ante): refactor `AnteHandler` (#1455)
- tests: add additional gas consumption tests (#1477)
- chore: update proto make commands (#1471)
- Remove unbound labels from added custom tendermint metrics (#1434)
- deps: bump SDK to v0.46.6 (#1486)
- fear(eip712): Add EIP-712 encoding for multiple messages of the same type (#1483)
- fix: enable `fixIssue172` flag for non-deterministic keyring test (#1447)
- fix(tests): Delete inconsistent test (#1481)
- build(deps): bump alpine from 3.16.3 to 3.17.0 (#1492)
- build(deps): bump google.golang.org/grpc from 1.50.1 to 1.51.0 (#1490)
- build(deps): bump github.com/onsi/ginkgo/v2 from 2.5.0 to 2.5.1 (#1489)
- chore: refactor imports naming for clarity (#1491)
- fix: align empty account result for old blocks as ethereum (#1484)
- imp(ante): refactor AnteHandler (#1479)
- chore: Update linter and protogen configuration (#1478)
- fix(server): telemetry setup (#1497)
- chore(evm) - Delete deprecated store migrations (#1498)
- fix(evm): Added Cancun and Shanghai blocks to ChainConfig (#1499)
- chore(app): add store listener to Ethermint app (#1501)
- build(deps): bump cosmossdk.io/math from 1.0.0-beta.3 to 1.0.0-beta.4 (#1502)
- fix(app): register node service (#1505)
- fix(cmd): add missing GetAuxToFeeCommand (#1504)
- chore(feemarket): Delete deprecated migration logic (#1508)
- Remove simulation checks (#1507)
- chore: verify fees refactor (#1496)
- json-rpc(filters) fix block hash on newBlock filter (#1503)
- fix(ci): add gitleaks config (#1513)
